src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.ts

```diff
@@ -262,6 +271,23 @@ export class CodeEditorTutorAssessmentContainerComponent implements OnInit, OnDe
         }
     }
 
+    /**
+     * Load the feedback suggestions for the current submission from Athena.
+     */
+    private async loadFeedbackSuggestions(): Promise<void> {
+        this.feedbackSuggestions = (await firstValueFrom(this.athenaService.getFeedbackSuggestionsForProgramming(this.exerciseId, this.submission!.id!))) ?? [];
+        const allFeedback = [...this.referencedFeedback, ...this.unreferencedFeedback]; // pre-compute to not have to do this in the loop
+        // Don't show feedback suggestions that have the same description and reference - probably it is coming from an earlier suggestion anyway
+        this.feedbackSuggestions = this.feedbackSuggestions.filter((suggestion) => {
+            for (const feedback of allFeedback) {
+                if (feedback.detailText === suggestion.detailText && feedback.reference === suggestion.reference) {
+                    return false; // exclude this suggestion
+                }
+            }
+            return true; // include this suggestion
+        });
```

Comment:
We could leverage some of the array functions here. What do you think?

```suggestion
        this.feedbackSuggestions = this.feedbackSuggestions.filter((suggestion) =>
            allFeedback.every((feedback) =>
                feedback.detailText !== suggestion.detailText || feedback.reference !== suggestion.reference
            )
        );
```

Reply:
Nice suggestion! I added the change.